echo_server.py

```diff
@@ -0,0 +1,89 @@
+import socket
+
+
+def create_server_socket():
+    server_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+    # Set up the server socket
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+    return server_socket
+
+
+def response_ok():
+    return """\
+HTTP/1.1 200 OK\r\n\
+Content-Type: text/plain\r\n\r\n\
+Hey you.\r\n\
+"""
+
+
+def response_error(e):
+    return """\
+HTTP/1.1 {error}\r\n\
+\r\n\
+{cause}\r\n\
+    """.format(error=e.args[0], cause=e.args[1])
```

Comment:
This is closer to correct, but again, Python will add another `\n` to the end of each line.  Not what you want, really.